types/textarea-caret/index.d.ts

```diff
@@ -10,4 +10,6 @@ interface Options {
     debug?: boolean | undefined;
 }
 
-declare function textarea_caret(element: HTMLElement, position: number, options?: Options): Caret;
+declare global {
+    function getCaretCoordinates(element: HTMLElement, position: number, options?: Options): Caret;
+}
```

Comment:
The typical way to handle UMD files is to use the `export as namespace` syntax, like this:
```suggestion
declare function getCaretCoordinates(element: HTMLElement, position: number, options?: Options): Caret;

export as namespace getCaretCoordinates;
```

This makes the default export available in the global context without needing a `declare global` block.

Reply:
I tried it, but got this error from `pnpm test textarea-caret`:

```
dtslint@0.2.32
Error: 
/home/joeyparrish/work/DefinitelyTyped/types/textarea-caret/index.d.ts
  13:1  error  TypeScript@5.1, 5.2, 5.3, 5.4, 5.5, 5.6, 5.7, 5.8, 5.9 compile error:
Global module exports may only appear in module files  @definitelytyped/expect
```

Here's the exact diff I tried, in case I misunderstood you:

```diff
diff --git a/types/textarea-caret/index.d.ts b/types/textarea-caret/index.d.ts
index 3ceca331e2..96204e2769 100644
--- a/types/textarea-caret/index.d.ts
+++ b/types/textarea-caret/index.d.ts
@@ -1,5 +1,3 @@
-export = getCaretCoordinates;
-
 interface Caret {
     top: number;
     left: number;
@@ -10,6 +8,6 @@ interface Options {
     debug?: boolean | undefined;
 }
 
-declare global {
-    function getCaretCoordinates(element: HTMLElement, position: number, options?: Options): Caret;
-}
+declare function getCaretCoordinates(element: HTMLElement, position: number, options?: Options): Caret;
+
+export as namespace getCaretCoordinates;
```

Reply:
The `export = getCaretCoordinates;` line should still stay in the file too—the `export as namespace` works in conjunction with that to make a UMD module's type definitions.

That way, the module can either be imported with CommonJS or used as a global module depending on the context. It also prevents developers from accidentally trying to use the global name (i.e. `window.getCaretCoordinates`) in a context where it isn't defined, since TypeScript ignores `export as namespace` in source files that have imports.

Reply:
I can't manage to test the global name.  For example, I get this error:

`'getCaretCoordinates' refers to a UMD global, but the current file is a module. Consider adding an import instead  @definitelytyped/expect`

Should I skip testing this, or is there another way to do the test?

Reply:
Oh, never mind.  `window.` as a prefix works for testing the global.